### PR TITLE
Issue #126 NetworkImpl improvements for wi-fi toggling

### LINKED ISSUE #126: Update NetworkImpl
The current version doesn't turn off wi-fi because we don't have enough permissions.
There is a more stable version. 
The reference - /KISA_mobile/src/androidTest/java/com/common/network/NetworkUpgradedImpl.kt

### LINKED ISSUE #139: Support of Network in Android Q and higher
The considering interface is [Network](https://github.com/KasperskyLab/Kaspresso/blob/master/kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/device/network/Network.kt), the implementation is [NetworkImpl](https://github.com/KasperskyLab/Kaspresso/blob/master/kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/device/network/NetworkImpl.kt).

The problem: the used ```wifiManager.isWifiEnabled``` android method in the implementation is not supported in Android Q and higher.

The potential solution: Using Kautomator to toggle wi-fi manually without any calls of Android API.

### LINKED ISSUE #157: Changing state of WiFi/Mobile data breaks down the tests (Error: exitCode=137)
When I call the methods: 
```
toggleWiFi(true)
enable()
```
on Android Emulator with API level 26, it throws the following exception
`com.kaspersky.kaspresso.internal.exceptions.AdbServerException: command=shell svc wifi enable was performed with failed result=CommandResult(status=FAILED, description=exitCode=137, message=)`

**Solution:** I wrote two methods, that fix this problem:
```
fun Network.enableWifiAndMobileData() {
    runCatching {
        toggleWiFi(true)
        enable()
    }
}

fun Network.disableWifiAndMobileData() {
    runCatching {
        toggleWiFi(false)
        disable()
    }
}
```
**Note:** DON'T change order of called methods, because it breaks the work of the function

### PR DESCRIPTION
**Pull request by matzuk**
Closes #126
Closes #139 
Closes #157

### CODE REVIEW

File: kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/device/network/NetworkImpl.kt [hunk 1]
```diff
@@ -29,7 +32,8 @@ class NetworkImpl(
      * Required Permissions: INTERNET.
      */
     override fun disable() {
-        adbServer.performAdb("shell svc data disable", "shell svc wifi disable")
+        adbServer.performAdb("shell svc data disable")
```
[the remark this answers is not in the document]
Comment (matzuk) [reply kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/device/network/NetworkImpl.kt:35]: Do you suggest to include an additional method for network data?

[the remark this answers is not in the document]
Comment (RuslanMingaliev) [reply kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/device/network/NetworkImpl.kt:35]: y

File: kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.kaspersky.components.kautomator.component.switch.UiSwitch
+import com.kaspersky.components.kautomator.screen.UiScreen
+
+internal object NotificationScreen : UiScreen<NotificationScreen>() {
+
+    override val packageName: String = "com.android.systemui"
+
+    val wifiSwitcher = UiSwitch {
+        withContentDescription("Wi-Fi,Wifi signal full.,AndroidWifi")
```
Comment (RuslanMingaliev) [inline kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt:11]: Does it work with real devices?

File: kautomator/kautomator-framework/src/main/kotlin/com/kaspersky/components/kautomator/component/switch/UiSwitch.kt [hunk 3]
```diff
@@ -5,14 +5,15 @@ import com.kaspersky.components.kautomator.component.common.assertions.UiBaseAss
...
+import com.kaspersky.components.kautomator.component.text.UiTextViewAssertions
 
 /**
  * Ui View with [UiSwitchableActions] and [UiBaseAssertions]
  *
  * @see UiSwitchableActions
  * @see UiBaseAssertions
  */
-class UiSwitch : UiBaseView<UiSwitch>, UiSwitchableActions, UiBaseAssertions {
+class UiSwitch : UiBaseView<UiSwitch>, UiSwitchableActions, UiTextViewAssertions {
```
Comment (RuslanMingaliev) [inline kautomator/kautomator-framework/src/main/kotlin/com/kaspersky/components/kautomator/component/switch/UiSwitch.kt:16]: Are you sure all the method provided by `UiTextViewAssertions` work with `UiSwitch`?

File: kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt (hunk 2 above)
Comment (matzuk) [reply kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt:11]: Good question. I am going to research

File: kautomator/kautomator-framework/src/main/kotlin/com/kaspersky/components/kautomator/component/switch/UiSwitch.kt (hunk 3 above)
Comment (matzuk) [reply kautomator/kautomator-framework/src/main/kotlin/com/kaspersky/components/kautomator/component/switch/UiSwitch.kt:16]: I think, yes.
image

File: kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt (hunk 2 above)
Comment (matzuk) [reply kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt:11]: @RuslanMingaliev Nope, it so depends on a type of a device. On each device, the flow is different. I suggest describing these restrictions in the documentation of `Network` class and the throwing exception. What do you think?

Comment (RuslanMingaliev) [reply kaspresso/kaspresso-framework/src/main/kotlin/com/kaspersky/kaspresso/internal/outscreens/NotificationScreen.kt:11]: It doesn't work even with emulator, I tried Android 10 x86. 

> I suggest describing these restrictions in the documentation of Network class and the throwing exception.

I think any restrictions are evil. We should avoid them in any way.

**Pull request merged**